### PR TITLE
ASM Scaling UI

### PR DESCRIPTION
**Pull request by ellensi**
### Purpose

This is a first-cut functionality pull request for ASM Scaling UI implementation based on the discussion thread in [MAGN-10340](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-10340). Further UI enhancements have been proposed, but this pull request only reflects those specified in the above link.

A new dialog has been added to change the _scale factor_ that is dependent to the current DYN file. To select the working range, click and drag the vertical scroller (slider) up or down. In further enhancement, user will be able to move the selected range by dragging the highlight box as well as the scroller bar.

After user clicks OK, the change in geometry scaling and rendering is immediate but needs to be saved to replace the current DYN file. A new attribute is added in the source DYN file:

``` xml
<Workspace Version="1.1.1.1111"
           X="0" Y="0" zoom="1"
           ScaleFactor="100"
           Name="Home" Description=""
           RunType="Automatic" RunPeriod="1000"
           HasRunWithoutCrash="True" />
```

_Note:_ Changing the _scale factor_ will not take effect before LibG is updated to support this functionality.
### Screenshots

image

image

[[Older image link here]](https://cloud.githubusercontent.com/assets/6386550/17408133/9a99443a-5a9c-11e6-866d-a2d0b3c153c0.png)

In these two screenshots, the _scale factors_ will be **1** and **1000** respectively.
### Declarations
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] Snapshot of UI changes, if any.
### Reviewers

@aparajit-pratap
### FYIs

@Benglin

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -857,6 +870,10 @@ void _model_PropertyChanged(object sender, PropertyChangedEventArgs e)
                         this.PublishCurrentWorkspaceCommand.RaiseCanExecuteChanged();
                     RaisePropertyChanged("IsPanning");
                     RaisePropertyChanged("IsOrbiting");
+                    if (ChangeScaleFactorCommand != null)
```
Comment (ellensi) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: The *CanExecute* status of this new command still does not change properly in some situations (e.g. when the workspace is changed but there is no call to `RaiseProeprtyChanged("CurrentWorkspace")`. Will investigate for a fix after this PR's functionality has been confirmed.

**Comment by Benglin** [comment]
Hi @ellensi, that looks nice (have not reviewed yet). Just wondering if this is still needed? It seems to be targeting `master` branch instead of a side branch for scaling work.

**Comment by ikeough** [comment]
@Racel PTAL.

**Comment by Racel** [comment]
@ikeough @ellensi @Benglin - Please take a look at my comments in https://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-10340. I think the UI needs a few more adjustments. Please see my mockup (last comment with image). This is what I suggest we implement. It has a more linear flow, and I think will be easier to navigate. Let me know your thoughts...

**Comment by aparajit-pratap** [comment]
@Benglin this is very much still needed. We can't get it in just yet until the backend work in LibG is complete.

**Comment by aparajit-pratap** [comment]
Opened a new PR #7455 which is a clone of this one plus minor fixes.

**Pull request closed without merging**